teach/src/main/java/skype/teach/np/NpAddress.java

```diff
@@ -0,0 +1,58 @@
+package skype.teach.np;
+
+/**
+ * @author NPrilepa
+ *         Class Address override method Equals(Address adr)
+ */
+class Address {
+    private String cityName;
+    private String streetName;
+    private String streetType;
+    private int houseNum;
+    private int flatNum;
+    private int indexNum;
+
+    Address(String cityName, String streetName, String streetType, int houseNum, int flatNum, int indexNum) {
+        this.cityName = cityName;
+        this.streetName = streetName;
+        this.streetType = streetType;
+        this.houseNum = houseNum;
+        this.flatNum = flatNum;
+        this.indexNum = indexNum;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof Address)) {
+            return false;
+        }
+        Address addr = (Address) obj;
+        // indexNum not important for result
+        if (this.cityName.equals(addr.cityName)) {
```

Comment:
Any field can be unset. Valid address may contain only city.
So equals should return true if street is null in both addresses, for example.